lib/elixir/lib/list.ex

```diff
@@ -591,7 +591,12 @@ defmodule List do
   """
   @spec to_string(:unicode.charlist) :: String.t
   def to_string(list) when is_list(list) do
-    case :unicode.characters_to_binary(list) do
+    try do
+       :unicode.characters_to_binary(list)
+    rescue
+      ArgumentError ->
+        raise ArgumentError, "cannot make this list into a string (use 'inspect(list)' for its internal representation as a string)"
```

Comment:
Slept on it. How about this:

    must be a list of integers representing codepoints, of strings, or a list of such lists; got: [:a]

Sent from my iPhone

> On 13 Oct 2015, at 22:18, Aleksei Magusev <notifications@github.com> wrote:
> 
> In lib/elixir/lib/list.ex:
> 
> > @@ -591,7 +591,12 @@ defmodule List do
> >    """
> >    @spec to_string(:unicode.charlist) :: String.t
> >    def to_string(list) when is_list(list) do
> > -    case :unicode.characters_to_binary(list) do
> > +    try do
> > +       :unicode.characters_to_binary(list)
> > +    rescue
> > +      ArgumentError ->
> > +        raise ArgumentError, "cannot make this list into a string (use 'inspect(list)' for its internal representation as a string)"
> @henrik actually your last message looks better. 
> 
> —
> Reply to this email directly or view it on GitHub.
>